### DISCUSSION TITLE
issue: Using allowClear function of antd select components does not reset field.value

### DISCUSSION THREAD

**Discussion by SugaKaori** [react-hook-form/react-hook-form#9826] [Q&A] [answered]
Is this a bug?
Can you please confirm?

https://codesandbox.io/s/basic-usage-antd-4-24-7-forked-v38mt9?file=/demo.tsx

**Comment by Moshyfawn** [comment, marked as answer]
`Antd Select` doesn't return an event but the input value `onChange`. Additionally, it returns `undefined` when cleared, which is reserved by RHF for default values; `undefined` is not a valid value to pass into the field change handler.

The solution is to default the returned value to an empty string or `null`
```js
onChange={(value = '') => {
  field.onChange(value)
}}
```
